src/v/cluster/tests/data_migration_table_test.cc

```diff
@@ -83,6 +84,12 @@ create_groups(std::vector<std::string_view> strings) {
 
 struct data_migration_table_fixture : public seastar_test {
     ss::future<> SetUpAsync() override {
+        config::shard_local_cfg().cloud_storage_enabled.set_value(true);
+        config::shard_local_cfg()
+          .cloud_storage_disable_archiver_manager.set_value(true);
+        config::shard_local_cfg().cloud_storage_enable_remote_write.set_value(
+          true);
```

Comment:
this should be expressed as `config::mock_binding` passed as an argument to the table constructor

Reply:
used a boolean ctor argument as discussed